lms/models/grouping.py

```diff
@@ -86,6 +86,7 @@ class Type(str, Enum):
     parent_id = sa.Column(sa.Integer(), nullable=True)
     children = sa.orm.relationship(
         "Grouping",
+        foreign_keys=[parent_id, application_instance_id],
```

Comment:
This was missing from this other relationship. It's now mandatory because it becomes ambiguous with the new one.